views/settings/index.js

```diff
@@ -102,36 +140,106 @@ export default class SettingsView extends React.Component {
   }
 
   render() {
-    let loggedIn = this.state.success
-    let loading = this.state.loading
+    let loggedInGoogle = this.state.loggedInGoogle
+    let loggedInStOlaf = this.state.loggedInStOlaf
+    let loadingGoogle = this.state.loadingGoogle
+    let loadingStOlaf = this.state.loadingStOlaf
+    let username = this.state.username
+    let password = this.state.password
+    let disabledGoogle = loadingGoogle
+    let disabledStOlaf = loadingStOlaf
 
-    let disabled = loading
+    let loginTextStyleGoogle = disabledGoogle
+      ? styles.loginButtonTextDisabled
+      : loadingGoogle
+        ? styles.loginButtonTextLoading
+        : styles.loginButtonTextActive
 
-    let loginTextStyle = disabled
+    let loginTextStyleStOlaf = disabledStOlaf
       ? styles.loginButtonTextDisabled
-      : loading
+      : loadingStOlaf
         ? styles.loginButtonTextLoading
         : styles.loginButtonTextActive
 
     let actionCell = (
       <CustomCell
         contentContainerStyle={styles.actionButton}
-        isDisabled={disabled}
-        onPress={loggedIn ? this.logOut : this.logIn}
+        isDisabled={disabledGoogle}
+        onPress={loggedInGoogle ? this.logOutGoogle : this.logInGoogle}
+      >
+        <Text style={[styles.loginButtonText, loginTextStyleGoogle]}>
+          {loadingGoogle
+            ? 'Logging in to Google…'
+            : loggedInGoogle
+              ? 'Sign Out of Google'
+              : 'Sign In with Google'}
+        </Text>
+      </CustomCell>
+    )
+
+    let usernameCell = (
+      <CustomCell contentContainerStyle={styles.loginCell}>
+        <Text onPress={this.focusUsername} style={styles.label}>Username</Text>
```

Comment:
Gotcha. I've run out of time for the moment, but I'll be back on it later. Good catch.